examples/hcp-aks-demo/main.tf

```diff
@@ -33,43 +34,62 @@ module "network" {
   nsg_ids = { for i, subnet in keys(var.vnet_subnets) : subnet => azurerm_network_security_group.nsg.id }
 
   depends_on = [azurerm_resource_group.rg]
+
+  subnet_delegation = {
+    "subnet1" = {
+      "aks-delegation" = {
+        service_name = "Microsoft.ContainerService/managedClusters"
+        service_actions = ["Microsoft.Network/virtualNetworks/subnets/join/action"]
+      }
+    }
+  }
 }
 
 # Create an HCP HVN.
-resource "hcp_hvn" "hvn" {
-  cidr_block     = var.hvn_cidr_block
-  cloud_provider = "azure"
-  hvn_id         = var.hvn_id
-  region         = var.hvn_region
+# resource "hcp_hvn" "hvn" {
+#   cidr_block     = var.hvn_cidr_block
+#   cloud_provider = "azure"
+#   hvn_id         = var.hvn_id
+#   region         = var.hvn_region
+# }
+
+data "hcp_hvn" "example" {
+  hvn_id = "hvn-azure-westus2"
```

Comment:
note-to-self: replace data sources for HVN and HCP Consul cluster (used data source since we can create pre-release HCP consul clusters only via UI right now).